src/infer_response.cc

```diff
@@ -72,10 +78,20 @@ InferenceResponse::InferenceResponse(
     TRITONSERVER_InferenceResponseCompleteFn_t response_fn,
     void* response_userp,
     const std::function<
-        void(std::unique_ptr<InferenceResponse>&&, const uint32_t)>& delegator)
+        void(std::unique_ptr<InferenceResponse>&&, const uint32_t)>& delegator,
+    uint64_t seq_idx
+#ifdef TRITON_ENABLE_METRICS
+    ,
+    uint64_t infer_start_ns
+#endif  // TRITON_ENABLE_METRICS
+    )
     : model_(model), id_(id), allocator_(allocator), alloc_userp_(alloc_userp),
       response_fn_(response_fn), response_userp_(response_userp),
-      response_delegator_(delegator), null_response_(false)
+      response_delegator_(delegator), seq_idx_(seq_idx),
```

Comment:
What happen if the backend gets 10 response objects from the response factory, and the send them in the reverse order they are obtained from the factory?

Reply:
Updated

Reply:
Fixed